src/pubd/repository.rs

```diff
@@ -348,39 +667,46 @@ impl RrdpServer {
             // - a number that is higher than the current serial
             // - a number that is lower than the last delta (if set)
             if let Ok(serial) = u64::from_str(entry.file_name().to_string_lossy().as_ref()) {
-                trace!("Found serial: {}", serial);
-
                 // Skip the current serial
                 if serial == self.serial {
-                    trace!("Matches current serial, skipping");
                     continue;
                 // Clean up old serial dirs
                 } else if !self.notification.includes_delta(serial)
                     && !self.old_notifications.iter().any(|n| n.includes_delta(serial))
                 {
-                    info!("Deltas no longer contained, will delete: {}", path.to_string_lossy());
-                    if path.is_dir() {
+                    if config.retention_archive {
+                        // If archiving is enabled, then move these directories under the archive base
+
+                        let mut dest = self.rrdp_archive_dir.clone();
+                        dest.push(self.session.to_string());
+                        dest.push(format!("{}", serial));
+
+                        info!("Archiving RRDP serial '{}' to '{}", serial, dest.to_string_lossy());
+                        let _ = fs::create_dir_all(&dest);
+                        let _ = fs::rename(path, dest);
```

Comment:
I don't recall how the HTTP serving works, will it be impacted if a file is renamed out of the way while being served?